### PR TITLE
[KOGITO-9838] Add a troubleshooting guide for remote service invocations

### PR DESCRIPTION
**Pull request by ricardozanini**
**JIRA:**
https://issues.redhat.com/browse/KOGITO-9838

**Description:**
In this PR, we complement the work in KOGITO-9838 by adding a quick troubleshooting guide for remote service invocations. We should keep updating this guide with more information.

https://github.com/apache/incubator-kie-kogito-runtimes/pull/3238

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributions doc](https://github.com/apache/incubator-kie-kogito-docs/blob/main/CONTRIBUTING.md)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting main: `[0.9.x] KOGITO-XYZ Subject`
- [ ] The nav.adoc file has a link to this guide in the proper category
- [ ] The index.adoc file has a card to this guide in the proper category, with a meaningful description

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

**Review by krisv — APPROVED** [review]

**Review by fjtirado — COMMENTED** [review]
@ricardozanini 
I think having a troubleshooting guide is a must (so really good idea) and the content you write in no time is really good, but I would add a longer introduction and convert the current in just one section,  "hhtp troubleshooting", so current sections become sub-sections of that one. 
That way, in the future, we can add more sections, for example GRPC troubleshooting or JQ troubleshooting. 
wdyt?

**Review by tiagodolphine — APPROVED** [review]

**Review by fjtirado — APPROVED** [review]

**Review by MarianMacik — COMMENTED** [review]
Looks good, just minor suggestions.

File: serverlessworkflow/modules/ROOT/nav.adoc [hunk 1]
```diff
@@ -31,6 +31,7 @@
 *** xref:service-orchestration/configuring-openapi-services-endpoints.adoc[Advanced Configuration]
 *** xref:service-orchestration/working-with-openapi-callbacks.adoc[Callbacks]
 ** xref:service-orchestration/orchestration-of-grpc-services.adoc[gRPC]
+** xref:service-orchestration/troubleshooting.adoc[Troubleshooting]
```
[the remark this answers is not in the document]
Comment (ricardozanini) [reply serverlessworkflow/modules/ROOT/nav.adoc:34]: Based on @fjtirado suggestions (and honestly I thought about this before the PR), I'll make this guide more generic so we can add inner sections about overall troubleshooting services orchestrations.

**Comment by ricardozanini** [comment]
@fjtirado can you take a look?

**Review by MarianMacik — APPROVED** [review]
Looks good now, thanks!

**Review by wmedvede — APPROVED** [review]

**Pull request merged**